### ISSUE TITLE
Feature request: `tensor.round`

### ISSUE THREAD

**Issue by FlysoftBeta** [huggingface/candle#1025]
For now, I can't find an API like [PyTorch's](https://pytorch.org/docs/stable/generated/torch.round.html#torch.round) that can round elements inside a tensor.
Am I missing something or it just hasn't been implemented?

**Comment by LaurentMazare** [comment]
I don't think we have it yet indeed, could you give a bit more details about your use case for this so that we can check how much of the optional arguments we want to support (currently it's possible to do it in a very hacky way by doing `.to_dtype(I64).to_dtype(F32)` for example).

**Comment by FlysoftBeta** [comment]
> I don't think we have it yet indeed, could you give a bit more details about your use case

I have a model that produces a f32 tensor (for example, `[0.1,0.2,0.6,0.9]`). I use `.eq` to compare it with the target tensor (`[0, 0, 1, 1]`) to get the training accuracy. Now with `.to_dtype(u32)`, the output tensor becomes zeros `[0,0,0,0]`. So I need something to round it.

BTW, I am a newbie to machine learning, so my way to get the training accuracy may be bad.🥲

> so that we can check how much of the optional arguments we want to support

 I would like a argument to specify how many decimal places we should keep. 

> currently it's possible to do it in a very hacky way by doing .to_dtype(I64).to_dtype(F32) for example

Hmmm... Isn't the code dropping everything after the decimal point?

**Comment by LaurentMazare** [comment]
> > I don't think we have it yet indeed, could you give a bit more details about your use case
> 
> I have a model that produces a f32 tensor (for example, `[0.1,0.2,0.6,0.9]`). I use `.eq` to compare it with the target tensor (`[0, 0, 1, 1]`) to get the training accuracy. Now with `.to_dtype(u32)`, the output tensor becomes zeros `[0,0,0,0]`. So I need something to round it.
> 
> BTW, I am a newbie to machine learning, so my way to get the training accuracy may be bad.🥲

I guess this really depends on how your model work. The typical thing is to have a model that produces probabilities for each classes (through logits) and apply some `argmax` function on top of this which makes computing accuracy reasonably straightfowrard. You can see an example of this [here](https://github.com/huggingface/candle/blob/11d3687cc655f8f79d856342a5539a9274e96df4/candle-examples/examples/mnist-training/main.rs#L154-L159).

> > currently it's possible to do it in a very hacky way by doing .to_dtype(I64).to_dtype(F32) for example
> 
> Hmmm... Isn't the code dropping everything after the decimal point?

Right but you can add `0.5` before hand if you want to round to the nearest instead.

**Comment by FlysoftBeta** [comment]
> Just merged #1030 which adds the 3 usual rounding functions, `round`, `ceil`, and `floor`. Note that these don't take the number of decimal places as argument as I'm not entirely convinced of what the best api would look like to do this - in the meantime you can just do something like `(f * 100.)?.round()? * 0.01` to round with 2 digits (and have your helper function to do this if you repeat this pattern often enough).

Thanks for your work!😄
